src/battle/other_battle_calculators.c

```diff
@@ -907,6 +907,85 @@ u8 CalcSpeed(void *bw, struct BattleStruct *sp, int client1, int client2, int fl
     return ret;
 }
 
+/**
+ *  @brief Sorts clients' execution order factoring in who has already performed their action
+ *  @param bw battle work structure; void * because we haven't defined the battle work structure. Apparently we have but we don't use it here so
+ *  @param sp global battle structure
+ */
+void DynamicSortClientExecutionOrder(void *bw, struct BattleStruct *sp) {
+    int maxBattlers;
+    int i, j;
+    int temp1, temp2;
+    int currentAttackerId = 0;
+
+    maxBattlers = BattleWorkClientSetMaxGet(bw);
+
+    for (i = 0; i < maxBattlers; i++) {
+        if (sp->attack_client == sp->client_agi_work[i]) {
```

Comment:
may want to swap `turn_order` as well?  and may also want to add a parameter to the function that specifies the start client potentially (instead of always relying on `attack_client`)?